api/clients/v2/payload_disperser.go

```diff
@@ -0,0 +1,222 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/api/clients/codecs"
+	"github.com/Layr-Labs/eigenda/api/clients/v2/verification"
+	dispgrpc "github.com/Layr-Labs/eigenda/api/grpc/disperser/v2"
+	"github.com/Layr-Labs/eigenda/common/geth"
+	core "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	gethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// PayloadDisperser provides the ability to disperse payloads to EigenDA
+//
+// This struct is goroutine safe.
+type PayloadDisperser struct {
+	logger          logging.Logger
+	config          *PayloadDisperserConfig
+	codec           codecs.BlobCodec
+	disperserClient DisperserClient
+	certVerifier    verification.ICertVerifier
+}
+
+// BuildPayloadDisperser builds a PayloadDisperser from config structs
+func BuildPayloadDisperser(
+	logger logging.Logger,
+	payloadDisperserConfig *PayloadDisperserConfig,
+	disperserClientConfig *DisperserClientConfig,
+	// signer to sign blob dispersal requests
+	signer core.BlobRequestSigner,
+	// prover is used to compute commitments to a new blob during the dispersal process
+	//
+	// IMPORTANT: it is permissible for the prover parameter to be nil, but operating with this configuration
+	// puts a trust assumption on the disperser. With a nil prover, the disperser is responsible for computing
+	// the commitments to a blob, and the PayloadDisperser doesn't have a mechanism to verify these commitments.
+	//
+	// TODO: In the future, an optimized method of commitment verification using fiat shamir transformation will
+	//  be implemented. This feature will allow a PayloadDisperser to offload commitment generation onto the
+	//  disperser, but the disperser's commitments will be verifiable without needing a full-fledged prover
+	prover encoding.Prover,
+	accountant *Accountant,
+	ethConfig geth.EthClientConfig,
+) (*PayloadDisperser, error) {
+
+	codec, err := codecs.CreateCodec(
+		payloadDisperserConfig.PayloadPolynomialForm,
+		payloadDisperserConfig.BlobEncodingVersion)
+	if err != nil {
+		return nil, fmt.Errorf("create codec: %w", err)
+	}
+
+	disperserClient, err := NewDisperserClient(disperserClientConfig, signer, prover, accountant)
```

Comment:
this function feels like a mix between a dependency injection constructor and a builder. We have some configs but also need to inject 3 dependencies (signer, prover, accountant).

Feels like the DisperserClient could use a builder constructor

Reply:
> Feels like the DisperserClient could use a builder constructor

Agreed. I started implementing one based on this suggestion, but it requires broader changes (for example, creating a `DispserserClient` builder constructor would also entail creating an `AccountantConfig`)

What do you think of removing `BuildPayloadDisperser` and only providing a constructor with dependency injection for now? Doing so would remove the need to consider `Accountant` and `Prover` entirely from this struct, relying instead on that `DisperserClient` passed in.

Reply:
Think that would make sense. I would need to play with the code to myself get a feeling for what the best way to make this work would be, so I'll defer to your sense of aesthetics here :)

Reply:
Simplified for now [f4a6f804](https://github.com/Layr-Labs/eigenda/pull/1159/commits/f4a6f804daec74338cd6d9e242270c4fc2c185f1)